test/integration/supervisor_login_test.go

```diff
@@ -230,6 +234,107 @@ func TestSupervisorLogin(t *testing.T) {
 			wantDownstreamIDTokenUsernameToMatch: regexp.QuoteMeta(env.SupervisorUpstreamLDAP.TestUserDN),
 			wantDownstreamIDTokenGroups:          env.SupervisorUpstreamLDAP.TestUserDirectGroupsCNs,
 		},
+		{
+			name: "activedirectory with all default options",
```

Comment:
It might be worth having another test in this test table which shows that you can override the all defaults with similar but different values.

Since specifying a search base is a very different behavior in the production code, it would be nice for that test to specify both search bases and make some assertions about what should happen.

Reply:
Since another important behavior different between generic LDAP and AD is the nested group search, it would be nice for the test that overrides the default to override in such a way that it disables nested group search. The test assertions about group membership could be based on the test env vars that already exist which distinguish between the expected direct and indirect group memberships.

Reply:
Resolved in cc3875f